src/core/metadata_manager.py

```diff
@@ -39,27 +39,26 @@
 from lib.timing import get_utc_now, global_start
 
 
-@Singleton
 class MetadataManager:
-    metadata_extractors: list[type(MetadataBase)] = []
-
-    blacklisted_for_cache = [MaliciousExtensions, ExtractFromFiles, Javascript]
-
     def __init__(self) -> None:
         self._logger = get_logger()
-        self._setup_extractors()
+        self.metadata_extractors: list[MetadataBase] = [
```

Comment:
Based on which reason did you move the definition of variables into the constructor? Do you intend to have only the constructor initialize variables?

Reply:
Two reasons: 
 - I am a big fan of Resource acquisition is initialization (RAII). Calling a function that may or may not succeed in the constructor may leave the whole class in a undefined state. Imagine a try catch in that function where the attribute may or may not be defined based on an exception being raised.
 - Principle of least surprise: As a developer I want to look at the head of the class and the constructor and see all fields being initialized there.

There is always the option (as used above) to call a function that returns the value that I can then assign in the constructor.